evm/src/executor/fork/backend.rs

```diff
@@ -420,15 +421,20 @@ impl SharedBackend {
     /// dropped.
     ///
     /// NOTE: this should be called with `Arc<Provider>`
-    pub fn spawn_backend<M>(provider: M, db: BlockchainDb, pin_block: Option<BlockId>) -> Self
+    pub fn spawn_backend<M>(
+        provider: M,
+        db: BlockchainDb,
+        pin_block: Option<BlockId>,
+        runtime: Option<Runtime>,
```

Comment:
hmm not a big fan of supplying the runtime here.
I'd consider creating a new runtime negligible additional effort

Reply:
Yes, however, I am establishing the IPC and WebSocket connections. You cannot use those in different runtimes or after the runtime that it was created on ceases to exist.

Reply:
imho this function should be async